### ISSUE TITLE
Why direct finetune in your paper is so slow?

### ISSUE THREAD

**Issue by ReproValid** [beichenzbc/Long-CLIP#104]
Hello, I am following up on your work. I tried to fine-tune CLIP directly using long texts, but my scores are significantly different from those in your paper. My scores are very high, whereas the ones in your paper are much lower. Why is there such a discrepancy? Could you share the details of your experiment?

**Comment by beichenzbc** [comment]
What's your position embedding interpolation strategy?

**Comment by wswaq** [comment]
> What's your position embedding interpolation strategy?

Sorry, I didn't perform interpolation on CLIP. I still have a few questions:

Aside from interpolation, are the other settings consistent with those in train.md?

Have you tried fine-tuning without interpolation?

**Comment by ReproValid** [comment]
Sorry, I didn't perform interpolation on CLIP. I still have a few questions:

Aside from interpolation, are the other settings consistent with those in train.md?

Have you tried fine-tuning without interpolation?

**Comment by beichenzbc** [comment]
1. DF in our paper represents liner interpolation + fine-tuning with only long captions. Other settings are consistent
2. We didn't try fine-tuning without interpolation since we want our model to support 200+ token input

**Comment by ReproValid** [comment]
Thanks!

**Comment by ReproValid** [comment]
Hello,  

I have interpolated and then fine-tuned CLIP on your codebase, removing the loss related to "short" and using only the ShareGPT-4V dataset. All other settings remain consistent with your code. However, my results are still significantly higher than those reported in your paper.  

Additionally, the paper *"CLIP-MOE: TOWARDS BUILDING MIXTURE OF EXPERTS FOR CLIP WITH DIVERSIFIED MULTIPLET UPCYCLING"* (https://arxiv.org/pdf/2409.19291) is also based on your work. I noticed that their results for directly fine-tuning CLIP on retrieval are much higher than what you reported, which aligns closely with my reproduction.  

Moreover, the ImageNet fine-tuning results in your paper seem overly low. My results are around 70. Could you please clarify if I have made any mistakes in my reproduction? If needed, I can share my code with you.  

Thank you!

**Comment by beichenzbc** [comment]
You should set the keep_length in the positional embedding stretching part into zero. We have two core methods: KPS and PCM. Only removing short loss function still uses KPS.

**Comment by ReproValid** [comment]
I followed your suggestion and applied linear interpolation with a coefficient of 3 to all the pos embedding. The results are now more reasonable, but they are still higher than the numbers reported in your paper. However, they align more closely with the reproduction results from "CLIP-MOE: TOWARDS BUILDING MIXTURE OF EXPERTS FOR CLIP WITH DIVERSIFIED MULTIPLET UPCYCLING" (https://arxiv.org/pdf/2409.19291).
Here is some results:
coco/image_retrieval_recall@1: 0.4296    coco/text_retrieval_recall@1: 0.5610    coco/image_retrieval_recall@5: 0.6910   coco/text_retrieval_recall@5: 0.8046    coco/image_retrieval_recall@10: 0.7854  coco/text_retrieval_recall@10: 0.8774   imagenet-zeroshot-val-top1: 0.6254      imagenet-zeroshot-val-top5: 0.8732
And the hyper params are set to your train.py with lr 1e-6 epochs 2,batch size 128 and i use 4 a100 80 gpus.
I think, after finetuning the retrieval results will be better.
